src/view/taffy_layout.rs

```diff
@@ -0,0 +1,149 @@
+// Copyright 2022 The Druid Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::{any::Any, marker::PhantomData};
+
+use vello::peniko::Color;
+
+use crate::view::{Id, VecSplice, ViewMarker, ViewSequence};
+use crate::widget::{self, ChangeFlags};
+use crate::MessageResult;
+
+use super::{Cx, View};
+
+/// TaffyLayout is a container view which does layout for the specified ViewSequence.
+///
+/// Children are positioned according to the Block, Flexbox or CSS Grid algorithm, depending
+/// on the display style set. If the children are themselves instances of TaffyLayout, then
+/// they can set styles to control how they placed, sized, and aligned.
+pub struct TaffyLayout<T, A, VT: ViewSequence<T, A>> {
+    children: VT,
+    style: taffy::Style,
+    background_color: Option<Color>,
+    phantom: PhantomData<fn() -> (T, A)>,
+}
+
+/// creates a vertical [`TaffyLayout`].
+pub fn flex_column<T, A, VT: ViewSequence<T, A>>(children: VT) -> TaffyLayout<T, A, VT> {
+    TaffyLayout::new_flex(children, taffy::FlexDirection::Column)
+}
+
+/// creates a horizontal [`TaffyLayout`].
+pub fn flex_row<T, A, VT: ViewSequence<T, A>>(children: VT) -> TaffyLayout<T, A, VT> {
+    TaffyLayout::new_flex(children, taffy::FlexDirection::Row)
+}
+
+/// creates a horizontal [`TaffyLayout`].
+pub fn div<T, A, VT: ViewSequence<T, A>>(children: VT) -> TaffyLayout<T, A, VT> {
```

Comment:
Div isn't a very compelling name. If there's an alternative you can think of, I'd like to change to it

Reply:
Hmm... it's called `div` because that's the name on web, and I can't think of why you'd want to use block layout unless you're emulating web. I could just remove it entirely? Or call it something like `raw_taffy_layout`?